docs/dev-environment-setup.md

```diff
@@ -0,0 +1,249 @@
+    ```bash
+     ~/.bashrc:
+         export PERL5LIB=~/lib/perl5/lib/perl5/
+
+         PERL_LOCAL_LIB_ROOT="/Users/pbastian/lib/perl5${PERL_LOCAL_LIB_ROOT:+:${PERL_LOCAL_LIB_ROOT}}"; export PERL_LOCAL_LIB_ROOT;
+         PERL_MB_OPT="--install_base \"/Users/pbastian/lib/perl5\""; export PERL_MB_OPT;
+         PERL_MM_OPT="INSTALL_BASE=/Users/pbastian/lib/perl5"; export PERL_MM_OPT;
+    ```
+
+## 5. Install pgTAP
+
+Ensure the postgres service is running with `pg_ctl start`.
+
+```
+cd schema && make install_pgtap
+```
+This clones and builds [pgTAP](https://pgtap.org/), the database unit testing library from source. The cloned repository will be left behind (visible with `git status`), which you can safely remove after the installation is complete.
+
+## 6. Deploy the Development Database
+
+```
+createdb ciip_portal_dev ciip_portal_test
+```
+
+`psql --list` to verify the databases were created.
+
+Rather than interacting with Sqitch directly, we use a [custom script](../schema/data/deploy-data.sh) for convenience to deploy the database, including seed data for dev, test and production.
+
+
+From the project root, run `.bin/deploy-data.sh --help` to see possible options. In development, this is most commonly run with `.bin/deploy-data.sh -d -dev` to drop the current database and redeploy after pulling down database-changing work from the remote or tampering with the database.
+
+## 7. Install and Run the Web Application
+
+Install the `node_modules` including `devDependencies` from the `package.json`:
+```
+cd app && yarn --dev
+```
+
+Run the development server with `yarn dev` and navigate in your browser to `localhost:3004`.
+
+Work that changes Relay GraphQL queries will require you to re-run `yarn build:relay` (which alternatively can be run using a `--watch` flag).
+
+### GraphQL Query Debugger
+
+Graphiql is an interactive in-browser GraphQL IDE available in development at `localhost:3004/graphiql`. It is handy for developing queries interactively before inserting them into a Relay fragment.
+
+### Voyager
+
+Voyager is an API graph explorer for GraphQL and is available in development at `localhost:3004/voyager`.
+
+## 7.5 Bypass Authentication in Development
+
+There are some shortcuts to bypass authentication in development and run the server while logged in as a preset user:
+
+```
+cd app && yarn dev <AS_USER>
+```
+...where `<AS_USER>` is one of the following:
+
+```
+AS_REPORTER
+AS_CERTIFIER
+AS_ANALYST
+AS_ADMIN
+```
+
+[Learn more here](./all-views-inventory.md) about the functionality available to each user type.
+
+## 8. Run Tests
+
+### Unit Tests with Jest
+
+```
+cd app && yarn test
+```
+Front-end unit tests are heavily snapshot-based. Work that changes the DOM will result in a diff from the last accepted snapshot and cause related tests to fail. You can update the snapshots and review / accept the diff with `yarn test -u`.
+
+### End-to-end Tests with Cypress
+
+```
+cd app && yarn cypress
```

Comment:
cypress requires that the server be running. `yarn cypress` by itself will start the cypress gui, but will show an error that no server is running.

Reply:
You're right, I guess that's not obvious. Adding, as well as a note to run with `yarn dev` and not bypassing auth.